docs/best-practices.md

```diff
@@ -0,0 +1,41 @@
+### How to select good environmental variable name?
+When selecting for environmental variable name, please check Dotnet cli convention examples.
+Please check discussion [here](https://github.com/NuGet/Client.Engineering/issues/882#issuecomment-816118683) for more details, please note this's not ultimate guide just our learning from last experience and hoping to help you, feel free to improve it. 
+Since env var name is case sensitive in Linux, so please use all uppercase (separated by underscore) naming convention for best practice.  But value doesn't have to case sensitive, we can convert to [uppercase](https://github.com/NuGet/NuGet.Client/blob/680f9bd4e97db7cd7482584276886764de69d3cb/src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs#L530) and compare to validate value.
```

Comment:
Consider an enumeration approach. Instead of writing full sentences, create a `Do` and `Don't` sections or just one bullet list. 

Ex a bullet point could be: 

* Use capitalized snake case for the environment variable name (ex. CAPITALIZED_SNAKE_CASE)

Reply:
Please check now.